apps/browser/qml/pages/components/PopUpMenu.qml

```diff
@@ -7,90 +7,177 @@
-            anchors.centerIn: parent
+    Component {
+        id: menuComponent
+
+        Item {
             width: popUpMenu.width
             height: popUpMenu.height
-            radius: popUpMenu.cornerRadius
-            visible: false
-        }
-    }
 
-    Rectangle {
-        width: popUpMenu.width
-        height: popUpMenu.height
-        color: Theme.colorScheme === Theme.LightOnDark ? "black" : "white"
-        radius: cornerRadius
-
-        ColorOverlay {
-            anchors.fill: parent
-            source: parent
-            color: Theme.primaryColor
-            opacity: Theme.opacityFaint
-        }
+            SilicaFlickable {
+                id: menuFlickable
+
+                width: popUpMenu.width
+                height: popUpMenu.height
+
+                contentHeight: contentLoader.y + contentLoader.height + footerLoader.height
+
+                interactive: popUpMenu.active   // Don't handle mouse events during fade out.
+
+                Private.AnimatedLoader {
+                    id: contentLoader
+
+                    x: menuFlickable.width - width - popUpMenu.margin
+                    y: Math.max(popUpMenu.margin, footerLoader.y - popUpMenu.margin - (Screen.sizeCategory > Screen.Medium
+                                ? container.contentHeight + footer.height
+                                : (isPortrait
+                                    ? Theme.paddingLarge * popUpMenu.heightRatio
+                                    : Screen.width - Theme.paddingMedium * 2)))
+
+                    width: footerLoader.width
+                    height: item ? item.height : 0
+                    source: popUpMenu.menuItem
 
-        SilicaFlickable {
-            id: container
-            width: parent.width
-            height: parent.height - footer.height
-            contentHeight: loader.height
-            clip: true
-            visible: popUpMenu.visible
-            onVisibleChanged: contentY = originY
-
-            Loader {
-                id: loader
-                width: parent.width
-                active: popUpMenu._open
-                sourceComponent: popUpMenu.menuItem
+                    onInitializeItem: {
+                        item.width = Qt.binding(function() { return footerLoader.width })
+                    }
+                }
+
+                children: [
+                    Rectangle {
+                        id: background
+
+                        x: contentLoader.x
+                        y: Math.max(popUpMenu.margin, contentLoader.y - menuFlickable.contentY)
+                        z: -1
+                        width: footerLoader.width
+                        height: footerLoader.y - y
+
+                        color: Qt.tint(
+                                    popUpMenu.palette.colorScheme === Theme.LightOnDark ? "black" : "white",
+                                    Theme.rgba(popUpMenu.palette.primaryColor, Theme.opacityFaint))
```

Comment:
I was a little unclear as to the purpose of the ColorOverlay item, and whether it was supposed to tint the entire window or just adjust the background color. If it is the latter then adjusting the background color like this is much more efficient, and if it's the former then a Rectangle with the color Theme.rgba(popUpMenu.palette.primaryColor, Theme.opacityFaint) should achieve the same effect as well.

Reply:
@adenexter ```ColorOverlay```  just adjust the background color

Reply:
Thanks, this should be okay as is then.